aten/src/ATen/nn_parse.py

```diff
@@ -418,12 +415,22 @@ def run(paths):
                 if cname + suffix in header_functions:
                     bwd_functions.append(header_functions[cname + suffix])
 
-            base = base_declaration(func, fwd_function, backends)
-            declarations.append(forward_declaration(base, fwd_function))
-            declarations.append(backward_declaration(base, bwd_functions))
+            defualt_scalar_types = ['Float', 'Double', 'Half']
+            forward_backend_types = {}
+            backward_backend_types = {}
+            for backend in backends:
+                backend_props = func.get(backend, {})
+                forward_backend_types[backend] = backend_props.get('forward_scalar_types', defualt_scalar_types)
+                backward_backend_types[backend] = backend_props.get('backward_backend_types', defualt_scalar_types)
```

Comment:
I don't see this being used anywhere in your diffs.  And in fact it looks like you are using 'backward_scalar_types'?

Splitting up diffs like this is fine, but it would be better if you include an actual working example in the diff.  This could be an existing specification where you just enter the defaults, but it's pretty difficult to figure out what's going on here without an example.

Reply:
Added example